service-proxy.js

```diff
@@ -918,6 +961,7 @@ function ensurePeerDisconnected(serviceName, peer, reason, now) {
 
     var peerServices = self.connectedPeerServices[peer.hostPort];
     if (!peerServices || isObjectEmpty(peerServices)) {
+        delete self.peersToConnect[peer.hostPort];
```

Comment:
is this sufficient ? The peer connector could be in it's "100ms setImmediate loop" right now. i.e. `peersToConnect` was already copied and its doing the smeared loop.